File: app/src/official/java/dolphin/android/apps/CpblCalendar/EmptyFragmentWithCallbackOnResume.java
package dolphin.android.apps.CpblCalendar;

import android.app.Activity;
import android.support.v4.app.Fragment;

/**
 * Created by dolphin on 2013/7/15.
 * <p/>
 * https://code.google.com/p/android/issues/detail?id=23096#c5
 * http://stackoverflow.com/a/12681526
 */
public class EmptyFragmentWithCallbackOnResume extends Fragment {
    public final static String TAG = "EmptyFragmentWithCallbackOnResume";

    OnFragmentAttachedListener mListener = null;

    @Override
    public void onAttach(Activity activity) {
        super.onAttach(activity);
        try {
            mListener = (OnFragmentAttachedListener) activity;
        } catch (ClassCastException e) {
            throw new ClassCastException(activity.toString()
                    + " must implement OnFragmentAttachedListener");
        }
    }

    @Override
    public void onResume() {
        super.onResume();
        if (mListener != null) {
            mListener.OnFragmentAttached();
        }
    }

    public interface OnFragmentAttachedListener {
        void OnFragmentAttached();
    }
}

File: app/src/flexible/java/dolphin/android/apps/CpblCalendar/EmptyFragmentWithCallbackOnResume.kt
package dolphin.android.apps.CpblCalendar

import android.content.Context
import androidx.fragment.app.Fragment

/**
 * https://code.google.com/p/android/issues/detail?id=23096#c5
 * http://stackoverflow.com/a/12681526
 */
class EmptyFragmentWithCallbackOnResume : Fragment() {

    private var mListener: OnFragmentAttachedListener? = null

    override fun onAttach(activity: Context) {
        super.onAttach(activity)
        try {
            mListener = activity as OnFragmentAttachedListener
        } catch (e: ClassCastException) {
            throw ClassCastException(activity.toString() + " must implement OnFragmentAttachedListener")
        }

    }

    override fun onResume() {
        super.onResume()
        mListener?.onFragmentAttached()
    }

    internal interface OnFragmentAttachedListener {
        fun onFragmentAttached()
    }

    companion object {
        val TAG = "EmptyFragmentWithCallbackOnResume"
    }
}
